stacks-signer/src/stacks_client.rs

```diff
@@ -158,143 +188,189 @@ impl StacksClient {
+        todo!("Get the next possible nonce from the stacks node");
+    }
 
-        let mut tx_signer = StacksTransactionSigner::new(&unsigned_tx);
-        tx_signer
-            .sign_origin(&self.stacks_private_key)
-            .map_err(|_| ClientError::SignatureGenerationFailure)?;
-        if let (Some(payer), Some(_)) = (payer, payer_nonce) {
-            tx_signer
-                .sign_sponsor(payer)
-                .map_err(|_| ClientError::SponsorSignatureGenerationFailure)?;
+    /// Helper function to retrieve the pox contract address and name from the stacks node
+    fn get_pox_contract(&self) -> Result<(StacksAddress, ContractName), ClientError> {
+        // Check if we have overwritten the pox contract ID in the config
+        if let Some(pox_contract) = self.pox_contract_id.clone() {
+            return Ok((pox_contract.issuer.into(), pox_contract.name));
         }
-
-        let Some(tx) = tx_signer.get_tx() else {
-            return Err(ClientError::SignatureGenerationFailure);
+        // TODO: we may want to cache the pox contract inside the client itself (calling this function once on init)
+        let send_request = || {
+            self.stacks_node_client
+                .get(self.pox_path())
+                .send()
+                .map_err(backoff::Error::transient)
         };
+        let response = retry_request(send_request)?;
+        if !response.status().is_success() {
+            return Err(ClientError::RequestFailure(response.status()));
+        }
+        let json_response = response.json::<serde_json::Value>()?;
+        let entry = "contract_id";
+        let contract_id_string = json_response
+            .get(entry)
+            .and_then(|id: &serde_json::Value| id.as_str())
+            .ok_or_else(|| ClientError::InvalidJsonEntry(entry.to_string()))?;
+        let id = QualifiedContractIdentifier::parse(contract_id_string).unwrap();
+        Ok((id.issuer.into(), id.name))
+    }
 
-        Ok(tx.serialize_to_vec())
+    /// Helper function that attempts to deserialize a clarity hex string as the aggregate public key
+    fn parse_aggregate_public_key(&self, hex: &str) -> Result<Option<Point>, ClientError> {
+        let public_key_clarity_value = ClarityValue::try_deserialize_hex_untyped(hex)?;
+        if let ClarityValue::Optional(optional_data) = public_key_clarity_value.clone() {
+            if let Some(ClarityValue::Sequence(SequenceData::Buffer(public_key))) =
+                optional_data.data.map(|boxed| *boxed)
+            {
+                let mut bytes = [0_u8; 32];
+                bytes.copy_from_slice(&public_key.data);
+                Ok(Some(Point::from(Scalar::from(bytes))))
+            } else {
+                Ok(None)
+            }
+        } else {
+            Err(ClientError::MalformedClarityValue(public_key_clarity_value))
+        }
+    }
+
+    /// Sends a transaction to the stacks node for a modifying contract call
+    #[allow(dead_code)]
```

Comment:
Yes this is still required because we are not yet actually submitting a modifying contract call anymore (originally we were because of the cast_aggregate_public_key_vote() function that I made, but we may no longer need this. This function will then only be used when making the sBTC transactions.